source/AsteroidField.cpp

```diff
@@ -125,10 +125,8 @@ void AsteroidField::Draw(DrawList &draw, const Point &center, double zoom) const
 
 
 // Check if the given projectile collides with any asteroids. This excludes minables.
-const vector<Collision> &AsteroidField::CollideAsteroids(const Projectile &projectile) const
+vector<Collision> &AsteroidField::CollideAsteroids(const Projectile &projectile, vector<Collision> &result) const
```

Comment:
this really does not need a return type, you are already handing in a reference, no need to return it when you can just push changes directly into it

Reply:
I made it return the list so it would be easier to use in for loops. It's not necessary, but requires less code changes overall.

Reply:
that one more line wont hurt anybody